src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -162,29 +157,46 @@ async def load_lockfile(
     python_setup: PythonSetup,
 ) -> LoadedLockfile:
     lockfile = request.lockfile
-    if isinstance(lockfile, Lockfile):
+    # TODO: Fold "resource://" URL support into the DownloadFile primitive, instead of
+    #  manually handling it here. That would also give us support for https:// URLs for tool
+    #  lockfiles (e.g., we could choose to download the default_lockfile_url instead of
+    #  embedding the lockfiles as resources). This would require capturing the SHA256 of
+    #  every current tool lockfile, and we need to think through the consequences of
+    #  downloading lockfiles, so we punt for now.
+    parts = urlparse(lockfile.url)
+    # urlparse retains the leading / in URLs with a netloc.
+    lockfile_path = parts.path[1:] if parts.path.startswith("/") else parts.path
+    if parts.scheme in {"", "file"}:
         synthetic_lock = False
-        lockfile_path = lockfile.file_path
         lockfile_digest = await Get(
             Digest,
             PathGlobs(
                 [lockfile_path],
                 glob_match_error_behavior=GlobMatchErrorBehavior.error,
-                description_of_origin=lockfile.file_path_description_of_origin,
+                description_of_origin=lockfile.url_description_of_origin,
             ),
         )
         _digest_contents = await Get(DigestContents, Digest, lockfile_digest)
         lock_bytes = _digest_contents[0].content
-    else:
+    elif parts.scheme == "resource":
         synthetic_lock = True
-        _fc = lockfile.file_content
+        _fc = FileContent(
+            lockfile_path,
+            # The "netloc" in our made-up "resource://" scheme is the package.
```

Comment:
isn't this comment stating the obvious here?

Reply:
Not really. I think the code reads weird (it's not obvious what the first argument to read_binary() is supposed to be without looking it up, but it's important to emphasize that this is not an actual network location).